packages/desktop-client/src/components/reports/spreadsheets/default-spreadsheet.tsx

```diff
@@ -1,61 +1,64 @@
 import * as d from 'date-fns';
 
-import q, { runQuery } from 'loot-core/src/client/query-helpers';
+import { runQuery } from 'loot-core/src/client/query-helpers';
 import { send } from 'loot-core/src/platform/client/fetch';
 import * as monthUtils from 'loot-core/src/shared/months';
-import { integerToAmount, amountToInteger } from 'loot-core/src/shared/util';
-
-import { index, indexStack } from '../util';
-
-export default function createSpreadsheet(
+import { integerToAmount } from 'loot-core/src/shared/util';
+import {
+  type AccountEntity,
+  type PayeeEntity,
+  type CategoryEntity,
+  type RuleConditionEntity,
+  type CategoryGroupEntity,
+} from 'loot-core/src/types/models';
+
+import {
+  categoryLists,
+  groupBySelections,
+  type UncategorizedEntity,
+  type UncategorizedGroupEntity,
+  type QueryDataEntity,
+} from '../ReportOptions';
+
+import filterHiddenItems from './filterHiddenItems';
+import makeQuery from './makeQuery';
+import recalculate from './recalculate';
+
+export type createSpreadsheetProps = {
+  start: string;
+  end: string;
+  categories: { list: CategoryEntity[]; grouped: CategoryGroupEntity[] };
+  selectedCategories: CategoryEntity[];
+  conditions: RuleConditionEntity[];
+  conditionsOp: string;
+  hidden: boolean;
+  uncat: boolean;
+  groupBy?: string;
+  balanceTypeOp?: string;
+  payees?: PayeeEntity[];
+  accounts?: AccountEntity[];
+  setDataCheck?: (value: boolean) => void;
+};
+
+export default function createSpreadsheet({
   start,
   end,
-  groupBy,
-  balanceTypeOp,
   categories,
   selectedCategories,
-  payees,
-  accounts,
   conditions = [],
   conditionsOp,
   hidden,
   uncat,
+  groupBy,
+  balanceTypeOp,
+  payees,
+  accounts,
   setDataCheck,
-) {
-  let uncatCat = {
-    name: 'Uncategorized',
-    id: null,
-    uncat_id: '1',
-    hidden: 0,
-    offBudget: false,
-  };
-  let uncatTransfer = {
-    name: 'Transfers',
-    id: null,
-    uncat_id: '2',
-    hidden: 0,
-    transfer: false,
-  };
-  let uncatOff = {
-    name: 'OffBudget',
-    id: null,
-    uncat_id: '3',
-    hidden: 0,
-    offBudget: true,
-  };
+}) {
+  let catList: CategoryEntity[] | UncategorizedEntity[];
+  let catGroup: CategoryGroupEntity[] | UncategorizedGroupEntity[];
 
-  let uncatGroup = {
-    name: 'Uncategorized',
-    id: null,
-    hidden: 0,
-    categories: [uncatCat, uncatTransfer, uncatOff],
-  };
-  let catList = uncat
-    ? [...categories.list, uncatCat, uncatTransfer, uncatOff]
-    : categories.list;
-  let catGroup = uncat
-    ? [...categories.grouped, uncatGroup]
-    : categories.grouped;
+  [catList, catGroup] = categoryLists({ hidden, uncat, categories });
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏TS should be able to automatically infer the return type of the function. If it's not doing so - this will likely point to an issue in the implementation.

```suggestion
  let [catList, catGroup] = categoryLists({ hidden, uncat, categories });
```

(changing the return statement of `categoryList` to `return [something, something] as const` might solve the problem for you)